### PR TITLE
807 - IdsMultiselect Additional Features/Fixes

### LINKED ISSUE #807: IdsMultiselect: Additional Features/Fixes
**Describe the bug**
The multiselect needs a little love so it works closer to https://main-enterprise.demo.design.infor.com/components/multiselect/example-index.html

**Tasks**
At least the following (may sync more tasks:

- [x] Keyboard Support
- [x] tags example is opening in wrong position
- [x] styling is off compared to old version (including groups)
- [x] doesn't stay open when selecting
- [x] Recheck #742 #363 
- [x] TODOs.md

### LINKED ISSUE #872: idsMultiselect: Misaligned checkbox label
**Describe the bug**
Misaligned checkbox label

**To Reproduce**

1. https://main.wc.design.infor.com/ids-multiselect/example.html
2. switch to dark or high contrast
3. open dropdown
4. see error
5. notice that the checkbox label is not centered

**Expected behavior**
Checkbox should be centered

**Version**

- ids-enterprise: [e.g. v4.9.0 or v4.10.0]

**Screenshots**
https://www.screencast.com/t/PFl1jqQOB

**Platform**
 - OS Version: [iOS 8]
 - Browser Name: [Safari]

**Additional context**
Add any other context about the problem here. NONE

### PR DESCRIPTION
**Pull request by andriivarhanov**
**Explain the details for making this change. What existing problem does the pull request solve?**
This PR includes `ids-multiselect` additional features/fixes:

- keyboard support
- styling
- popup stays opened when selecting
- groups
- side by side example
- other small adjustments and fixes

**Related github/jira issue (required)**:
Closes #807
Closes #872 

**Steps necessary to review your pull request (required)**:
- pull and build
- go to http://localhost:4300/ids-multiselect/example.html
- focus and open a multiselect with keyboard (arrow down, enter)
- select/unselect an option
- close with keyboard (excape)
- http://localhost:4300/ids-multiselect/groups.html
- check grouping functionality (select all items from a group and see group label doesn't appear)
- check tags and checkboxes styling in light/dark/contrast theme
- go to http://localhost:4300/ids-multiselect/side-by-side.html
- check side by side example

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
~- [ ] A note to the change log.~

### CODE REVIEW

**Review by EdwardCoyle — DISMISSED** [review]
Looks really nice @andriivarhanov 👍

File: src/components/ids-dropdown/ids-dropdown.ts [hunk 1]
```diff
@@ -12,15 +12,18 @@ import '../ids-icon/ids-icon';
 
 import styles from './ids-dropdown.scss';
 
-type IdsListBoxOption = {
+export type IdsListBoxOption = {
```
Comment (tmcconechy) [inline src/components/ids-dropdown/ids-dropdown.ts:15]: Wondering if this should be IdsDropdownOption (or go in IdsListBox but i think these are more specific options for dropdown)

**Review by tmcconechy — APPROVED** [review]
Looks good just one comment on the type. Nice work !

File: src/components/ids-dropdown/ids-dropdown.ts (hunk 1 above)
Comment (andriivarhanov) [reply src/components/ids-dropdown/ids-dropdown.ts]: makes sense, I've changed the naming

**Review by deep7102 — APPROVED** [review]
Looks nice 🍌 

FYI - Saw typo, [`HdsPopupElementRef`](https://github.com/infor-design/enterprise-wc/blob/f85dba41d5ef70d5d835970fa8f43ee01693cd2b/src/components/ids-popup/ids-popup.ts#L614) It's with main branch, so not specific to this PR.

**Review by tmcconechy — APPROVED** [review]

**Pull request merged**